### PR TITLE
Revisions of SPED report, now Student Report

### PR DESCRIPTION
**Pull request by jhilde**
This is the revisions of the SPED report which is now branded as the Student Report. 

Since this is my first PR, would appreciate feedback, especially on testing strategy. I'm also a verbose commenter it turns out, let me know if we'd prefer fewer comments.

Note that the report will have strange data for absences and tardies because of dummy data issues articulated in #835.

Here's a sample of the document generated: [sped_referral.pdf](https://github.com/studentinsights/studentinsights/files/788565/sped_referral.pdf)

### CODE REVIEW

**Comment by alexsoble** [comment]
Cool, awesome @jhilde and congrats on your first PR! Will get back to you on this with a first look tomorrow and a deeper dive on Thursday! 

🎉 🎉 🎉

File: app/views/students/sped_referral.pdf.erb [hunk 1]
```diff
@@ -1,61 +1,192 @@
...
-<p>
-  <h4>Current services:</h4>
-  <ul style="list-style-type:none">
+<style>
+
+* {
+  font-family: 'Open Sans', sans-serif !important;
+}
+
+.data-table  {border-collapse:collapse;border-spacing:0;border-color:#aaa;}
```
Comment (kevinrobinson) [inline app/views/students/sped_referral.pdf.erb]: I'd vote for each rule on its own line like this:

```
.data-table  {
  border-collapse:collapse;
  border-spacing:0;
  border-color:#aaa;
}
``

if you're cool with that.  Makes it easier to edit than super wide lines, and you can see +/- rules on diffs easier.

File: app/views/students/sped_referral.pdf.erb [hunk 2]
```diff
@@ -1,61 +1,192 @@
...
+<br style="clear: left;" /><br/>
+
+<!-- Services -->
+<% right = false %>
+<div>
+<h4>Services:</h4>
+  <% if @services.any? %>
     <% @services.each do |service| %>
-      <li><%= "Since #{format_date(service.date_started)} - #{service.service_type.name}" %></li>
+      <div style="float: left; width:370px; padding:5px">
```
Comment (kevinrobinson) [inline app/views/students/sped_referral.pdf.erb]: I was wondering if it these were `display: inline-block`, would the layout wrap on its own, without having to set and clear floats manually?  Or maybe there's something whack with the way CSS works in the PDF gem.

**Comment by kevinrobinson** [comment]
@jhilde nice!  This looks great 👍 

One bit of UX feedback is that I found the two-column layout harder to parse, particularly to figure out the order of events.  But that's always easy to change around if that's confusing to actual users, so no reason to hold up shipping anything.

I left two quick comments for you, and will leave it to you and @alexsoble to merge and 🚢 this.  Excited to see how folks use this, awesome work! :)

**Comment by alexsoble** [comment]
Thoughts: 

+ I think the code comments are great @jhilde! 
+ I also super appreciate the way you expanded the spec alongside expanding the feature.
+ Agreed with @kevinrobinson that we should find a solution that works with CSS only and doesn't require ruby variables like `<% right = false %>` ...
+ ... but I also don't think that should hold us back from merging and shipping this excellent work. Merging now!

**Comment by kevinrobinson** [comment]
💥 awesome!

**Comment by jhilde** [comment]
Beautiful!

On Wed, Feb 22, 2017 at 9:36 AM Kevin Robinson

<
mailto:Kevin Robinson <notifications@github.com>
> wrote:

a, pre, code, a:link, body { word-wrap: break-word !important; }

💥

awesome!

—

You are receiving this because you were mentioned.

Reply to this email directly,
https://github.com/studentinsights/studentinsights/pull/862#issuecomment-281686437
, or

**Pull request merged**